patsy/util.py

```diff
@@ -45,18 +45,13 @@
     _pandas_is_categorical_dtype = None
 else:
     if hasattr(pandas, "api"):
-        # This is available starting in pandas v0.19.0
+        # The pandas `api` submodule debuted with pandas v0.19. The helper
+        # method `is_categorical_dtype` exists as supported API prior to pandas
+        # 2.1.0, but internally it was equivalent to `isinstance(x,
+        # pandas.CategoricalDType)` for our use-cases. Since this is now the
+        # preferred code pathway, we use it always.
         have_pandas_categorical_dtype = True
-        _pandas_is_categorical_dtype = getattr(pandas.api.types, "is_categorical_dtype", None)
-        # pandas.api.types._pandas_is_categorical_dtype is deprecated in pandas v2.1.0
-        if _pandas_is_categorical_dtype is not None:
-            import warnings
-            with warnings.catch_warnings(record=True) as w:
-                _pandas_is_categorical_dtype(int)
-                if len(w) > 0 and issubclass(w[-1].category, FutureWarning):
-                    _pandas_is_categorical_dtype = None
-        else:
-            _pandas_is_categorical_dtype = lambda x: isinstance(x, pandas.CategoricalDType)
+        _pandas_is_categorical_dtype = lambda x: isinstance(x, pandas.CategoricalDType)
```

Comment:
Pretty sure this will not work on old pandas, e.g., those that work with python 2.7